pkgs/development/python-modules/pytest-pycodestyle/default.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  pycodestyle,
+  py,
+  pytest,
+  pytest-isort,
+}:
+
+buildPythonPackage rec {
+  pname = "pytest-pycodestyle";
+  version = "2.3.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "henry0312";
+    repo = "pytest-pycodestyle";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ODbjjAEGI1qkK4UXQA2q4Yvi53zP/vS1Ue50uCaJ4To=";
+  };
+
+  build-system = [ setuptools ];
+  dependencies = [ py pycodestyle ];
```

```suggestion
  dependencies = [ py pycodestyle ];
  buildInputs = [ pytest ];
```

To fix 
```
pytest-pycodestyle> Checking runtime dependencies for pytest_pycodestyle-2.3.1-py3-none-any.whl
pytest-pycodestyle>   - pytest not installed
```

while not propagating it to downstream packages

---

Adding pytest will cause a error with the `py.io` module, see: https://github.com/pytest-dev/pytest/issues/12606

---

The suggested change is to make up for https://github.com/NixOS/nixpkgs/pull/326978#discussion_r1685722153